presto-hive/src/main/java/com/facebook/presto/hive/metastore/file/FileHiveMetastore.java

```diff
@@ -270,15 +270,15 @@ else if (table.getTableType().equals(EXTERNAL_TABLE.name())) {
     }
 
     @Override
-    public Optional<Map<String, HiveColumnStatistics>> getTableColumnStatistics(String databaseName, String tableName, Set<String> columnNames)
+    public Map<String, HiveColumnStatistics> getTableColumnStatistics(String databaseName, String tableName, Set<String> columnNames)
     {
-        return Optional.of(ImmutableMap.of());
+        return ImmutableMap.of();
     }
 
     @Override
-    public Optional<Map<String, Map<String, HiveColumnStatistics>>> getPartitionColumnStatistics(String databaseName, String tableName, Set<String> partitionNames, Set<String> columnNames)
+    public Map<String, Map<String, HiveColumnStatistics>> getPartitionColumnStatistics(String databaseName, String tableName, Set<String> partitionNames, Set<String> columnNames)
     {
-        return Optional.of(ImmutableMap.of());
+        return ImmutableMap.of();
```

Comment:
There is a semantics change here.
Previously `SemiTransactionalHiveMetastore` wrapping `FileHiveMetastore` would see "no stats" and return "no stats" from `getPartitionColumnStatistics(String databaseName, String tableName, Set<String> partitionNames, Set<String> columnNames)`. Now it will see "can't calc" and proceed to  `partitionNamesToQuery.build().forEach(partionName -> resultBuilder.put(partionName, ImmutableMap.of()));` code path...

Is this change deliberate?
Does it matter? (I think it doesn't, but double-checking..)

Reply:
That's the entire point of this commit. It doesn't matter, but it introduces yet another abstraction to think about. Why did i introduce it? Because i had no idea what to return if no columns have statistics computed. `Optional.empty()` ? `Optional.of(ImmutableMap.of())`?